servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -7996,6 +8027,171 @@ public DomainList listDomainDependencies(String service) {
         }
     }
 
+    void executeDeleteExpiredMembership(ResourceContext ctx, ObjectStoreConnection con, String domainName, String roleName,
+                                        String normalizedMember, Timestamp expiration, String auditRef, String caller) {
+
+        final String principal = getPrincipalName(ctx);
+
+        // process our delete role member operation
+
+        if (!con.deleteExpiredRoleMember(domainName, roleName, normalizedMember, principal, expiration, auditRef)) {
+            throw ZMSUtils.notFoundError(caller + ": unable to delete role member: " +
+                    normalizedMember + " from role: " + roleName + ". this happened either because the member was deleted already or his expiration was updated very recently", caller);
+        }
+    }
+
+    List<ExpiryMember> executeDeleteDomainExpiredRoleMemberships(ResourceContext ctx, String domainName, List<ExpiryMember> members, String auditRef, String caller) {
+        List<ExpiryMember> removedList = new ArrayList<>();
+        for (int retryCount = defaultRetryCount; ; retryCount--) {
```

Comment:
because this is not an end user operation, let's not complicate the code with the retry logic. we should just try once and if it fails, just log it and move on since during next run we'll try to clean up again